elastic/kibana
# [Index Management] Add enrich policies fetch api and expose in plugin api

Related to https://github.com/elastic/kibana/pull/163159

## Summary

In order to unblock the esql team working on the kibana implementation, I'll first merge the API changes for listing enrich policies and a client side public api interface for exposing this mechanism so that the consumer plugins can use it.

Consumers of this plugin can use public `apiService` from index management by first adding the `indexManagement` plugin as a dependency of their plugin and then proceding to invoke the method as:

```typescript
const { data, error } = await indexManagement?.apiService.getAllEnrichPolicies();
```

Which will then return a serialized list of enriched policies that look something like:

```json
[
    {
      "name": "my-policy",
      "type": "match",
      "sourceIndices": ["users"],
      "matchField": "email",
      "enrichFields": ["first_name", "last_name", "city"]
    },
  ]
```

##### How to test
1. Start elasticsearch with `yarn es snapshot --license=trial` and serverless search solution with `yarn start`
2. Open the [ILM/public/plugin.ts](https://github.com/elastic/kibana/blob/main/x-pack/plugins/index_lifecycle_management/public/plugin.tsx#L41) file in your text editor and add `indexManagement?.apiService.getAllEnrichPolicies();`
3. Go to console and create a test enrich policy
4. Open ILM  policies page and make sure that a request for fetching `enrich policies` has been fired

<details>
<summary>Create a policy</summary>

```
PUT /users
{
  "mappings": {
    "properties": {
      "email": {
        "type": "keyword"
      },
      "first_name": {
        "type": "text"
      },
      "last_name": {
        "type": "text"
      },
      "city": {
        "type": "text"
      },
      "zip": {
        "type": "keyword"
      },
      "state": {
        "type": "keyword"
      }
    }
  }
}


PUT /_enrich/policy/my-policy
{
  "match": {
    "indices": "users",
    "match_field": "email",
    "enrich_fields": ["first_name", "last_name", "city", "zip", "state"]
  }
}
```
</details>
